common/web/types/src/keyman-touch-layout/keyman-touch-layout-file-reader.ts

```diff
@@ -0,0 +1,80 @@
+import Ajv from "ajv";
+import { TouchLayoutFile } from "./keyman-touch-layout-file.js";
+
+export class TouchLayoutFileReader {
+  public read(source: Uint8Array): TouchLayoutFile {
+    const decoder = new TextDecoder('utf-8', {fatal:true, ignoreBOM: true});
+    let sourceString: string;
+    try {
+      sourceString = decoder.decode(source);
+    /* c8 ignore next 7 */
+    } catch(e) {
+      if(e instanceof TypeError) {
+        // TODO: Do we want to do something else with this?
```

Comment:
Maybe, it's an unrecoverable error in any event though

Reply:
Unrecoverable but we want to turn it into a friendly compiler error rather than a compiler crash